code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -767,27 +773,42 @@
 
 // This proc attempts to find a new ruler to lead the hive.
 /datum/hive_status/proc/update_ruler()
-	if(living_xeno_ruler)
+	if(isxenoqueen(living_xeno_ruler))
 		return //No succession required.
 
 	var/mob/living/carbon/xenomorph/successor
+	var/list/mob/living/carbon/xenomorph/prio_candidates = xenos_by_tier[XENO_TIER_FOUR]
+	var/list/mob/living/carbon/xenomorph/seco_candidates = xenos_by_tier[XENO_TIER_THREE]
 
-	// TO DO: this shouldn't be a strict type check and should account for strains
-	var/list/candidates = xenos_by_typepath[/datum/xeno_caste/queen] + xenos_by_typepath[/datum/xeno_caste/shrike] + xenos_by_typepath[/datum/xeno_caste/king] + xenos_by_typepath[/datum/xeno_caste/dragon] + xenos_by_typepath[/datum/xeno_caste/king/conqueror]
-	if(length(candidates)) //Priority to the queens.
-		successor = candidates[1] //First come, first serve.
+	if(prio_candidates.len > 0)
+		for(var/mob/living/carbon/xenomorph/potential_successor in prio_candidates)
+			if(isxenoqueen(potential_successor))
+				successor = potential_successor
+				break
+			else
+				successor = potential_successor
```

Comment:
```suggestion
		for(var/mob/living/carbon/xenomorph/potential_successor in prio_candidates)
			if(isxenoqueen(potential_successor))
				successor = potential_successor
				break
			else
				successor = potential_successor
```
if(thing)
  x
   break
else
   x
  just do x immediately?

Reply:
because prio_candidates can be queen / shrike / king and it will always loop through them all. I want it to break once we have a queen. If no queen then just assign either shrike / king